code-challenges.test.js

```diff
@@ -15,24 +15,88 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+// describe("removeAndShuffle", () => {
+//   it("removes the first item from the array and shuffles the remaining items.", () => {
+//     const result1 = removeAndShuffle(colors1);
+//     const result2 = removeAndShuffle(colors2);
+//     expect(result1).toEqual(
+//       expect.arrayContaining(["yellow", "blue", "pink", "green"])
+//     );
+//     expect(result2).toEqual(
+//       expect.arrayContaining([
+//         "saffron",
+//         "aquamarine",
+//         "periwinkle",
+//         "indigo",
+//         "ochre",
+//       ])
+//     );
+//   });
+// });
+//  ReferenceError: removeAndShuffle is not defined
 
+const colors1 = ["purple", "blue", "green", "yellow", "pink"];
+// // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+const colors2 = [
+  "chartreuse",
+  "indigo",
+  "periwinkle",
+  "ochre",
+  "aquamarine",
+  "saffron",
+];
+// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// Create dynamic fxn by using any given array as a parameter
+// Destructuring to remove the first item
+// Loop through the rest of the array in reverse order
+// Generate a random index between 0 and i
+// Shuffling the array using Fisher-Yates
+// Swap the elements at indexes i and j
+// Return the shuffled array
+
+const removeAndShuffle = (arr) => {
+  const [, ...rest] = arr;
+  for (let i = rest.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [rest[i], rest[j]] = [rest[j], rest[i]]; // Shuffling the array using Fisher-Yates algorithm
+  }
+  return rest;
+};
+// console.log(removeAndShuffle(colors1));
+// log 1 = [ 'yellow', 'pink', 'green', 'blue' ]
+// log 2 = [ 'green', 'blue', 'pink', 'yellow' ]
+// console.log(removeAndShuffle(colors2));
+// log 1 = [ 'saffron', 'periwinkle', 'indigo', 'aquamarine', 'ochre' ]
+// log 2 = [ 'ochre', 'periwinkle', 'saffron', 'aquamarine', 'indigo' ]
```

Comment:
Beautiful code.  Excellent pseudocode. Just remember that semi-colons are no longer used at the end of lines. :)
Awesome job! 🎉

Reply:
Thanks Trish! I’m not sure why, but my option+shift+F formats my code with the semi colons at the end of my lines. I’ll try to figure that out!